### PR TITLE
Add XTZ address format support

### PR DESCRIPTION
**Pull request by krivenkoviktor1954**
Add support for Tezos (XTZ) addresses tz1, tz2, tz3 and KT1. Doc http://tezos.gitlab.io/api/p2p.html

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -239,6 +239,46 @@ if(!eos.PublicKey.isValid(data)) {
   return eos.PublicKey(data).toBuffer();
 }
 
+function tezosAddressEncoder(data: Buffer): string {
```
Comment (Arachnid) [inline src/index.ts]: Can you reuse the `makeBase58CheckEncoder` and `makeBase58CheckDecoder` functions? These functions seem to duplicate their functionality.

Comment (kaliumxyz) [reply src/index.ts]: Tezos addresses are not p2pkh addresses and so cannot use that decoder.

Comment (Arachnid) [reply src/index.ts]: They don't need to be - the base58check encoder/decoder supports aribtrary prefixes. It should be compatible with this format; if it's not, it should be easy to make it compatible. The current code largely replicates that function.

Comment (makoto) [reply src/index.ts:242]: I think `makeBase58CheckEncoder`/`makeBase58CheckDecoder` names are just confusing as the majority of the code were checking the optcode sequence of p2pkh/p2h. The only things they share are that they call bs58check.encode/decode (which are now called `bs58Decode/bs58Encode` ).  If these names are confusing, `makeBase58Check*` should just be renamed into `makeBitcoinBase58Check*`.

Comment (Arachnid) [reply src/index.ts:253]: @makoto They're not specific to p2pkh - they should work with any prefix.

**Pull request closed without merging**